java/gameserverSDK/src/main/java/com/microsoft/azure/gaming/HeartbeatThread.java

```diff
@@ -222,8 +227,11 @@ public void run() {
 
                 // Perform the operation that the server requested
                 PerformOperation(sessionInfo.getOperation());
+            } catch (RetryException | ExecutionException e) {
+                Logger.Instance().LogError("Checked exception occurred when sending a heartbeat to the agent", e);
```

Comment:
it's not clear how this loop exits with these changes.  If the agent is completely unreachable/gone/stuck, how does this loop exit?  

Presumably a RetryException indicates that the agent is present, and a retry will hopefully work.  It's less clear what causes an ExecutionException, and if that's a terminal state (ie the agent won't come back from that)